Failure in cpp2v with templates

Reported by @aa755.

@aa755 I needed to add `git submodule init --update --recursive` to get things to build, but I'm seeing a lot of build failures. Are these expected? I am building with clang22 rather than clang19, so that could be the source of the issue?

```
[203/861] /usr/bin/clang++ -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FIBER_NO_LIB -DBOOST_FIBER_STATIC_LINK -DJSON_HAS_RANGES=0 -DQUILL_ROOT_LOGGER_ONLY -D_GNU_SOURCE -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/third_party -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/concurrentqueue -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/ankerl -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/komihash -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/BLAKE3/c -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/ethash/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/evmc/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/quill/quill/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/unordered_dense/include -isystem /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/fiber/include -march=haswell -O2 -g -DNDEBUG -std=gnu++23 -Wall -Wextra -Wconversion -Werror -MD -MT category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o -MF category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o.d -o category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o -c /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp
FAILED: [code=1] category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o 
/usr/bin/clang++ -DBOOST_CONTEXT_DYN_LINK -DBOOST_CONTEXT_NO_LIB -DBOOST_FIBER_NO_LIB -DBOOST_FIBER_STATIC_LINK -DJSON_HAS_RANGES=0 -DQUILL_ROOT_LOGGER_ONLY -D_GNU_SOURCE -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/third_party -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/concurrentqueue -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/ankerl -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/komihash -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/BLAKE3/c -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/ethash/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/evmc/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/quill/quill/include -I/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/unordered_dense/include -isystem /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/third_party/fiber/include -march=haswell -O2 -g -DNDEBUG -std=gnu++23 -Wall -Wextra -Wconversion -Werror -MD -MT category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o -MF category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o.d -o category/mpt/CMakeFiles/monad_trie.dir/update_aux.cpp.o -c /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:17:
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/storage_pool.hpp:129:21: error: call to 'start_lifetime_as_array' is ambiguous
  129 |                     start_lifetime_as_array<std::atomic<uint32_t>>(
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_construct.h:347:5: note: candidate function [with _Tp = std::atomic<unsigned int>]
  347 |     start_lifetime_as_array(void* __p, size_t __n) noexcept
      |     ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/core/detail/start_lifetime_as_polyfill.hpp:65:15: note: candidate function [with T = std::atomic<unsigned int>]
   65 |     inline T *start_lifetime_as_array(void *p, size_t) noexcept
      |               ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:151:24: error: 'name' marked 'override' but does not override any member functions
  151 |     virtual string_ref name() const noexcept override
      |                        ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:160:35: error: 'payload_info' marked 'override' but does not override any member functions
  160 |     virtual base_::payload_info_t payload_info() const noexcept override
      |                                   ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:206:65: error: '_generic_code' marked 'override' but does not override any member functions
  206 |     virtual BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::generic_code _generic_code(
      |                                                                 ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:215:24: error: '_do_message' marked 'override' but does not override any member functions
  215 |     virtual string_ref _do_message(
      |                        ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:215:24: error: 'monad::async::sender_errc_with_payload_code_domain_::_do_message' hides overloaded virtual function [-Werror,-Woverloaded-virtual]
/usr/include/boost/outcome/experimental/status-code/status_code_domain.hpp:569:55: note: hidden overloaded virtual function 'system_error2::status_code_domain::_do_message' declared here: type mismatch at 1st parameter ('_vtable_message_args &' vs 'const system_error2::status_code<void> &')
  569 |   BOOST_OUTCOME_SYSTEM_ERROR2_CONSTEXPR20 virtual int _do_message(_vtable_message_args &args) const noexcept = 0;
      |                                                       ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:246:5: error: variable type 'const sender_errc_with_payload_code_domain_' is an abstract class
  246 |     sender_errc_with_payload_code_domain;
      |     ^
/usr/include/boost/outcome/experimental/status-code/status_code_domain.hpp:558:55: note: unimplemented pure virtual method '_do_name' in 'sender_errc_with_payload_code_domain_'
  558 |   BOOST_OUTCOME_SYSTEM_ERROR2_CONSTEXPR20 virtual int _do_name(_vtable_name_args &args) const noexcept = 0;
      |                                                       ^
/usr/include/boost/outcome/experimental/status-code/status_code_domain.hpp:560:56: note: unimplemented pure virtual method '_do_payload_info' in 'sender_errc_with_payload_code_domain_'
  560 |   BOOST_OUTCOME_SYSTEM_ERROR2_CONSTEXPR20 virtual void _do_payload_info(_vtable_payload_info_args &args) const noexcept = 0;
      |                                                        ^
/usr/include/boost/outcome/experimental/status-code/status_code_domain.hpp:567:56: note: unimplemented pure virtual method '_do_generic_code' in 'sender_errc_with_payload_code_domain_'
  567 |   BOOST_OUTCOME_SYSTEM_ERROR2_CONSTEXPR20 virtual void _do_generic_code(_vtable_generic_code_args &args) const noexcept = 0;
      |                                                        ^
/usr/include/boost/outcome/experimental/status-code/status_code_domain.hpp:569:55: note: unimplemented pure virtual method '_do_message' in 'sender_errc_with_payload_code_domain_'
  569 |   BOOST_OUTCOME_SYSTEM_ERROR2_CONSTEXPR20 virtual int _do_message(_vtable_message_args &args) const noexcept = 0;
      |                                                       ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:24:
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/detail/db_metadata.hpp:288:21: error: call to 'start_lifetime_as' is ambiguous
  288 |             return *start_lifetime_as<std::atomic<uint8_t>>(
      |                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_construct.h:288:5: note: candidate function [with _Tp = std::atomic<unsigned char>]
  288 |     start_lifetime_as(void* __p) noexcept
      |     ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/core/detail/start_lifetime_as_polyfill.hpp:41:15: note: candidate function [with T = std::atomic<unsigned char>]
   41 |     inline T *start_lifetime_as(void *p) noexcept
      |               ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:24:
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/detail/db_metadata.hpp:638:22: error: call to 'start_lifetime_as' is ambiguous
  638 |         auto *intr = start_lifetime_as<db_metadata>(buffer);
      |                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/bin/../lib64/gcc/x86_64-pc-linux-gnu/16.1.1/../../../../include/c++/16.1.1/bits/stl_construct.h:288:5: note: candidate function [with _Tp = monad::mpt::detail::db_metadata]
  288 |     start_lifetime_as(void* __p) noexcept
      |     ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/core/detail/start_lifetime_as_polyfill.hpp:41:15: note: candidate function [with T = monad::mpt::detail::db_metadata]
   41 |     inline T *start_lifetime_as(void *p) noexcept
      |               ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:67:17: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
   67 |                 typename StatusCode::domain_type().id() /* unique-ish based on domain's unique id */)
      |                 ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:183:55: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::indirecting_domain' requested here
  183 |   constexpr indirecting_domain<StatusCode, Allocator> _indirecting_domain{};
      |                                                       ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:187:12: note: in instantiation of variable template specialization 'system_error2::detail::_indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  187 |     return _indirecting_domain<StatusCode, Allocator>;
      |            ^
/usr/include/boost/outcome/experimental/status-code/status_code.hpp:566:72: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::get' requested here
  566 |       : _base(typename _base::_value_type_constructor{}, &domain_type::get(), static_cast<Args &&>(args)...)
      |                                                                        ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:214:12: note: in instantiation of function template specialization 'system_error2::status_code<system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>>::status_code<system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::payload_type *&>' requested here
  214 |     return status_code<domain_type>(in_place, p);
      |            ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:183:55: error: constexpr variable '_indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' must be initialized by a constant expression
  183 |   constexpr indirecting_domain<StatusCode, Allocator> _indirecting_domain{};
      |                                                       ^~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:187:12: note: in instantiation of variable template specialization 'system_error2::detail::_indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  187 |     return _indirecting_domain<StatusCode, Allocator>;
      |            ^
/usr/include/boost/outcome/experimental/status-code/status_code.hpp:566:72: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::get' requested here
  566 |       : _base(typename _base::_value_type_constructor{}, &domain_type::get(), static_cast<Args &&>(args)...)
      |                                                                        ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:214:12: note: in instantiation of function template specialization 'system_error2::status_code<system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>>::status_code<system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::payload_type *&>' requested here
  214 |     return status_code<domain_type>(in_place, p);
      |            ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:36:82: note: constructor of base class 'status_code_domain' is not called
   36 |   template <class StatusCode, class Allocator> class indirecting_domain : public status_code_domain
      |                                                                                  ^~~~~~~~~~~~~~~~~~
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:91:49: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
   91 |       return static_cast<status_code_domain &&>(typename StatusCode::domain_type())._do_name(args);
      |                                                 ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_name' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:103:49: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
  103 |       return static_cast<status_code_domain &&>(typename StatusCode::domain_type())._do_failure(c.value()->sc);
      |                                                 ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_failure' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:110:49: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
  110 |       return static_cast<status_code_domain &&>(typename StatusCode::domain_type())
      |                                                 ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_equivalent' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:118:42: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
  118 |       static_cast<status_code_domain &&>(typename StatusCode::domain_type())._do_generic_code(args2);
      |                                          ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_generic_code' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:126:58: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
  126 |       const int ret = static_cast<status_code_domain &&>(typename StatusCode::domain_type())._do_message(args2);
      |                                                          ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_message' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/update_aux.cpp:24:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/mpt/db_metadata_context.hpp:19:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/io.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/connected_operation.hpp:18:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/detail/connected_operation_storage.hpp:20:
In file included from /home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:23:
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:135:42: error: allocating an object of abstract class type 'typename system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>::domain_type' (aka 'monad::async::sender_errc_with_payload_code_domain_')
  135 |       static_cast<status_code_domain &&>(typename StatusCode::domain_type())._do_throw_exception(c.value()->sc);
      |                                          ^
/usr/include/boost/outcome/experimental/status-code/nested_status_code.hpp:65:15: note: in instantiation of member function 'system_error2::detail::indirecting_domain<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>::_do_throw_exception' requested here
   65 |     constexpr indirecting_domain() noexcept
      |               ^
/home/gmalecha/skylabs-ai/workspace-workspace-2/monad/category/async/sender_errc.hpp:313:51: note: in instantiation of function template specialization 'system_error2::make_nested_status_code<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>, monad::async::detail::thread_local_sender_errc_with_payload_code_allocator<system_error2::status_code<monad::async::sender_errc_with_payload_code_domain_>>>' requested here
  313 |     return BOOST_OUTCOME_SYSTEM_ERROR2_NAMESPACE::make_nested_status_code<
      |                                                   ^
17 errors generated.
```

--- a/rocq-skylabs-cpp2v/src/PrintDecl.cpp
+++ b/rocq-skylabs-cpp2v/src/PrintDecl.cpp
@@ -189,8 +189,7 @@ template <typename T> struct DeclPrinter {
     const Printer<T> print_body;
 
     DeclPrinter() = delete;
-    DeclPrinter(StringRef c, StringRef tc, Printer<T> p,
-                Valid v = alwaysValid)
+    DeclPrinter(StringRef c, StringRef tc, Printer<T> p, Valid v = alwaysValid)
         : ctor{c}, templated_ctor{tc}, invalid(v), print_body{p} {
         always_assert(p != nullptr);
     }
@@ -240,8 +239,9 @@ template <typename T> struct DeclPrinter {
         auto printDeclWith = [&]() {
             auto cp = cprint.withDecl(&decl);
             if (auto msg = invalid(box, ctxt)) {
-                guard::ctor _(print, print.templates() ? "Dtemplated_unsupported"
-                                                       : "Dunsupported");
+                guard::ctor _(print, print.templates()
+                                         ? "Dtemplated_unsupported"
+                                         : "Dunsupported");
                 cp.printName(print, decl) << fmt::nbsp;
                 print.str(msg);
                 return true;
@@ -683,16 +683,19 @@ static const char *supportedRecord(const CXXRecordDecl &decl,
         if (base.isVirtual())
             return "virtual base classes are not supported";
     }
-    return supportedRecord(static_cast<const RecordDecl&>(decl), ctxt);
+    return supportedRecord(static_cast<const RecordDecl &>(decl), ctxt);
 }
 
 } // namespace
 
-static const DeclPrinter Dstruct("Dstruct", "Dtemplated_struct", printStruct, supportedRecord);
-static const DeclPrinter Dunion("Dunion", "Dtemplated_union", printUnion, supportedRecord);
-static const DeclPrinter Drecord_struct("Dstruct", "Dtemplated_struct", printCStruct,
-                                        supportedRecord);
-static const DeclPrinter Drecord_union("Dunion", "Dtemplated_union", printCUnion, supportedRecord);
+static const DeclPrinter Dstruct("Dstruct", "Dtemplated_struct", printStruct,
+                                 supportedRecord);
+static const DeclPrinter Dunion("Dunion", "Dtemplated_union", printUnion,
+                                supportedRecord);
+static const DeclPrinter Drecord_struct("Dstruct", "Dtemplated_struct",
+                                        printCStruct, supportedRecord);
+static const DeclPrinter Drecord_union("Dunion", "Dtemplated_union",
+                                       printCUnion, supportedRecord);
 
 // Functions
 namespace {
@@ -937,8 +940,7 @@ static fmt::Formatter &printVar(CoqPrinter &print, const VarDecl &decl,
         return print.output() << "global_init.NoInit";
 }
 } // namespace
-static const DeclPrinter Dvariable("Dvariable", "Dtemplated_variable",
-                                   printVar,
+static const DeclPrinter Dvariable("Dvariable", "Dtemplated_variable", printVar,
                                    DeclPrinter<VarDecl>::alwaysValid);
 
 // Enumerations

--- a/rocq-skylabs-cpp2v/src/PrintDecl.cpp
+++ b/rocq-skylabs-cpp2v/src/PrintDecl.cpp
@@ -445,24 +445,18 @@ static fmt::Formatter &printStructBases(const CXXRecordDecl &decl,
         if (!type)
             fatal("null base class");
 
-        auto dependent = [&]() -> auto & {
-            guard::ctor _(print, "mkBase");
+        guard::ctor _(print, "mkBase");
+        if (auto record = type->getAsCXXRecordDecl()) {
+            cprint.printName(print, record, loc::of(type)) << fmt::nbsp;
+            auto li =
+                layout ? layout->getBaseClassOffset(record).getQuantity() : 0;
+            return printLayoutInfo(print, li);
+        } else {
             {
                 guard::ctor _2(print, "Ndependent'");
                 cprint.printType(print, type, loc::of(decl)) << fmt::nbsp;
             }
             return printLayoutInfo(print, 0);
-        };
-        if (print.templates())
-            return dependent();
-        else {
-            auto record = type->getAsCXXRecordDecl();
-            always_assert(record && "base class not a RecordType");
-            guard::ctor _(print, "mkBase");
-            cprint.printName(print, record, loc::of(type)) << fmt::nbsp;
-            auto li =
-                layout ? layout->getBaseClassOffset(record).getQuantity() : 0;
-            return printLayoutInfo(print, li);
         }
     });
 }

--- a/rocq-skylabs-cpp2v/tests/repro_250.t/Makefile
+++ b/rocq-skylabs-cpp2v/tests/repro_250.t/Makefile
@@ -1,0 +1,15 @@
+CPPFLAGS = -std=c++17
+
+CPP_FILES = $(shell find . -name '*.cpp')
+HPP_FILES = $(shell find . -name '*.hpp')
+
+GEN_FILES = $(CPP_FILES:.cpp=_cpp.v) $(HPP_FILES:.hpp=_hpp.v)
+
+Q = @
+
+all: $(GEN_FILES) $(GEN_VALCAT_FILES)
+
+%_cpp.v: %.cpp
+	$(Q)cpp2v -v -o $@ $< --check-types -- $(CPPFLAGS)
+%_hpp.v: %.hpp
+	$(Q)cpp2v -v -o $@ $< --check-types -- $(CPPFLAGS)

--- a/rocq-skylabs-cpp2v/tests/repro_250.t/run.t
+++ b/rocq-skylabs-cpp2v/tests/repro_250.t/run.t
@@ -1,0 +1,9 @@
+  $ . ../setup-project.sh
+
+Compiling the C++ code, use "make Q=" for debugging.
+  $ make
+  $ ls *.v | wc -l | sed -e 's/ //g'
+  1
+
+Compiling the generated Coq files.
+  $ dune build

--- a/rocq-skylabs-cpp2v/tests/repro_250.t/test.cpp
+++ b/rocq-skylabs-cpp2v/tests/repro_250.t/test.cpp
@@ -1,0 +1,3 @@
+template<class D> struct view_interface {};
+template<class T> struct chunk_view { struct OuterIter { struct value_type; }; };
+template<class T> struct chunk_view<T>::OuterIter::value_type : view_interface<value_type> {};

